src/WebJobs.Script/Binding/GeneralScriptBindingProvider.cs

```diff
@@ -0,0 +1,136 @@
+    {
+        public ITooling Tooling { get; set; }
+
+        public GeneralScriptBindingProvider(
+            JobHostConfiguration config, 
+            JObject hostMetadata, 
+            TraceWriter traceWriter) :
+            base(config, hostMetadata, traceWriter)
+        {
+        }
+
+        public override bool TryCreate(ScriptBindingContext context, out ScriptBinding binding)
+        {
+            string name = context.Type;
+            var attrType = this.Tooling.GetAttributeTypeFromName(name);
+            if (attrType == null)
+            {
+                binding = null;
+                return false;
+            }
+
+            var attrs = this.Tooling.GetAttributes(attrType, context.Metadata);
+                        
+            binding = new GeneralScriptBinding(this.Tooling, attrs, context);
+            return true;
+        }
+
+        public override bool TryResolveAssembly(string assemblyName, out Assembly assembly)
+        {
+            assembly = this.Tooling.TryResolveAssembly(assemblyName);
+            return (assembly != null);
+        }
+
+
+        // Context.DataType may frequently be missing. 
+        static Type GetRequestedType(ScriptBindingContext context)
+        {
+            Type type = ParseDataType(context);
+
+            if (type == null)
+            {
+                // DataType field is commonly missing. Default to JObject.  
+                return typeof(JObject);
+            }
+
+            Cardinality cardinality;
+            if (!Enum.TryParse<Cardinality>(context.Cardinality, out cardinality))
+            {
+                cardinality = Cardinality.One; // default 
+            }
+
+            if (cardinality == Cardinality.Many)
+            {
+                // arrays are supported for both trigger input as well
+                // as output bindings
+                type = type.MakeArrayType();
+            }
+            return type;
+
+        }
+
+        // Parse the DataType field and return as a System.Type.
+        private static Type ParseDataType(ScriptBindingContext context)
+        {
+            DataType result;
+            if (Enum.TryParse<DataType>(context.DataType, out result))
+            {
+                switch (result)
+                {
+                    case DataType.Binary:
+                        return typeof(byte[]);
+
+                    case DataType.Stream:
+                        return typeof(Stream);
+
+                    case DataType.String:
+                        return typeof(string);
+                }
+            }
+
+            return null; ;
+        }
+
+        class GeneralScriptBinding : ScriptBinding
+        {
+            private readonly Attribute[] _attributes;
+            private readonly ITooling _tooling;
+
+            private Type _defaultType;
+
+            public GeneralScriptBinding(ITooling tooling, Attribute[] attributes, ScriptBindingContext context)
```

Comment:
Are this the attributes for this binding only? Can you have a comment on this to make it a little clearer?